cf-execd/cf-execd.c

```diff
@@ -911,13 +977,52 @@ static bool ScheduleRun(EvalContext *ctx, Policy **policy, GenericAgentConfig *c
 
         GenericAgentConfigSetBundleSequence(config, NULL);
 
+#ifndef __MINGW32__
+        /* Take over the runagent_socket_allow_users set for comparison. */
+        StringSet *old_runagent_allow_users = NULL;
+        if (UsingRunagentSocket())
+        {
+            old_runagent_allow_users = (*execd_config)->runagent_allow_users;
+            (*execd_config)->runagent_allow_users = NULL;
+        }
+#endif
+
         *policy = LoadPolicy(ctx, config);
         ExecConfigDestroy(*exec_config);
         ExecdConfigDestroy(*execd_config);
 
         *exec_config = ExecConfigNew(!ONCE, ctx, *policy);
         *execd_config = ExecdConfigNew(ctx, *policy);
 
+#ifndef __MINGW32__
+        if (UsingRunagentSocket())
+        {
+            /* Check if the old list and the new one differ. */
+            if (!StringSetIsEqual(old_runagent_allow_users,
+                                  (*execd_config)->runagent_allow_users))
+            {
```

Comment:
If I'm being nitpicky: This comment would not help me much. Right now, it says the same as the code. For it to be helpful, I think it needs to mention that the policy changed (not the runagent sockets/dirs/files). For example:
```suggestion
            /* Check if the list of allowed users, from policy, has changed. */
            if (!StringSetIsEqual(old_runagent_allow_users,
                                  (*execd_config)->runagent_allow_users))
            {
```

Reply:
I think it's enough to look at where that list is taken from, just a few lines above.